ksql-serde/src/main/java/io/confluent/ksql/serde/delimited/KsqlDelimitedSerdeFactory.java

```diff
@@ -60,4 +74,23 @@ public KsqlDelimitedSerdeFactory() {
 
     return deserializer;
   }
+
+  private static class SchemaValidator implements SchemaWalker.Visitor<Void, Void> {
+
+    public Void visitPrimitive(final Schema schema) {
+      // Primitive types are allowed.
+      return null;
+    }
+
+    public Void visitBytes(final Schema schema) {
+      // Decimal type is allowed.
+      return null;
```

Comment:
we should explicitly check for Decimal with `DecimalUtil#isDecimal`

Reply:
Meh, the `schema` here is coming from KSQL, which would only pass bytes if its a decimal, so its a redundant change IMHO. Also, we should be looking to switch to our own schema soon and at which point this becomes a mute point.

If you feel strongly about this, I'll add it, but I think its just noise.

Reply:
I'll add it anyway